### ISSUE TITLE
RFC: use jemalloc or other alternative allocator

### ISSUE THREAD

**Issue by grigorig** [libass/libass#243]
libass does a lot of allocations. I did a bit of ad-hoc testing by LD_PRELOADing jemalloc into libass and profiling two simple script. I see 2-3% improvement with a dynamic script (animated, so lots of bitmap rendering) and around 4% with a static script.

Is this worth the effort? Adding optional jemalloc support mostly boils down to build system changes.

**Comment by wm4** [comment]
This is a bit of a problem, because some libass API uses actually require you to malloc/free data that is free/malloced by libass. Though that also means that using libass can severely break on Windows if libass and API user link different libcs.

I find this a bit silly though. Wouldn't it be better to remove the allocations which make it so slow?

**Comment by grigorig** [comment]
Yes, you are right. We could probably only use jemalloc for internal allocations that are never supposed to be freed by API users. This was mostly an off-beat idea that I was having some morning. I think if there weren't any side effects this could be a suitable band-aid solution to improve performance.

A better solution for libass should be to either get rid of allocations completely or to use purpose-tailored suballocators, when possible.
